### PR TITLE
Provide ability to pre-conflate scanned advertisements

### LINKED ISSUE #654: Android ANR when collecting advertisements
Still trying to figure out this one. I've been running into an ANR that I seem to consistently hit after anywhere from 15-45 minutes of successful scanning. I've been testing on an Galaxy S22 running Android 13. 

For context, I have an ongoing BLE scan running in a foreground service to connect to certain devices as they appear. This scan is meant to run alongside the app, indefinitely. I have only observed this when I leave the phone unplugged with the screen off for a while (with the foreground service still running in the background).

The ANR is as follows:

```Cmd line: <package>
Build fingerprint: 'samsung/b0qsqw/b0q:13/TP1A.220624.014/S908USQS2BWA2:user/release-keys'
ABI: 'arm64'
Build type: optimized
suspend all histogram:	Sum: 41.270ms 99% C.I. 0.494us-948.800us Avg: 78.311us Max: 2379us
DALVIK THREADS (168):
"main" prio=5 tid=1 TimedWaiting
  | group="main" sCount=1 ucsCount=0 flags=1 obj=0x7334ef58 self=0xb4000079a6fc6380
  | sysTid=6202 nice=0 cgrp=background sched=0/0 handle=0x7b521c3500
  | state=S schedstat=( 427635411952 12382667868 187478 ) utm=34641 stm=8121 core=2 HZ=100
  | stack=0x7fceb7f000-0x7fceb81000 stackSize=8188KB
  | held mutexes=
  at jdk.internal.misc.Unsafe.park(Native method)
  - waiting on an unknown object
  at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
  at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:88)
  at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
  at kotlinx.coroutines.BuildersKt.runBlocking(unavailable:1)
  at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
  at kotlinx.coroutines.BuildersKt.runBlocking$default(unavailable:1)
  at kotlinx.coroutines.channels.ChannelsKt__ChannelsKt.trySendBlocking(Channels.kt:38)
  at kotlinx.coroutines.channels.ChannelsKt.trySendBlocking(unavailable:1)
  at com.juul.kable.BluetoothLeScannerAndroidScanner$advertisements$1$callback$1.onBatchScanResults(BluetoothLeScannerAndroidScanner.kt:54)
  at android.bluetooth.le.BluetoothLeScanner$BleScanCallbackWrapper$2.run(BluetoothLeScanner.java:681)
  ...
```

This can be followed to this line in `BluetoothLeScannerAndroidScanner.kt:54`:

https://github.com/JuulLabs/kable/blob/b69d0ff84e0a5e1075c695d35bb7de32f364280d/core/src/androidMain/kotlin/BluetoothLeScannerAndroidScanner.kt#L54

### PR DESCRIPTION
**Pull request by djweber**
It was observed in #654 that using the default blocking send under certain conditions on various devices could result in ANRs being triggered when performing scans.  This PR provides a way for users to opt in to make the Android scanner's `callbackFlow` use a non-blocking `trySend` instead of `trySendBlocking`, with no change to the existing behavior.

Closes #654

### CODE REVIEW

**Review by twyatt — APPROVED** [review]

**Comment by twyatt** [comment]
@djweber add a commit to your branch with some small changes, otherwise looks great! Thanks for the contribution. I'll merge after I get another PR approval.

**Review by cedrickcooke — APPROVED** [review]
Lgtm

**Comment by twyatt** [comment]
Thanks for the contribution @djweber!
Shipped w/ [0.32.0](https://github.com/JuulLabs/kable/releases/tag/0.32.0).

**Pull request merged**
